cider/exec/template/DateTimeTranslator.cpp

```diff
@@ -117,6 +117,53 @@ std::string from_datetrunc_field(const DatetruncField& fieldno) {
 
 }  // namespace
 
+ExtractField ExtractExpr::presto_function_to_extract_field(
+    const std::string& function_name) {
+  ExtractField fieldno;
+  if (boost::iequals(function_name, "year")) {
+    fieldno = kYEAR;
+  } else if (boost::iequals(function_name, "quarter")) {
+    fieldno = kQUARTER;
+  } else if (boost::iequals(function_name, "month")) {
+    fieldno = kMONTH;
+  } else if (boost::iequals(function_name, "day")) {
+    fieldno = kDAY;
+  } else if (boost::iequals(function_name, "quarterday")) {
+    fieldno = kQUARTERDAY;
+  } else if (boost::iequals(function_name, "hour")) {
+    fieldno = kHOUR;
+  } else if (boost::iequals(function_name, "minute")) {
+    fieldno = kMINUTE;
+  } else if (boost::iequals(function_name, "second")) {
+    fieldno = kSECOND;
+  } else if (boost::iequals(function_name, "millisecond")) {
+    fieldno = kMILLISECOND;
+  } else if (boost::iequals(function_name, "microsecond")) {
```

Comment:
use map better?

Reply:
I previously considered changing input to lower case and use a map to compare. But I see lots of large casewhen, if in our code, but I haven't seen somewhere using a map.

Reply:
just a suggestion. I read some codes. Ref
https://github.com/intel/BDTK/blob/af0dd42b977332488f33cf1b0984a606707a78e3/cider/exec/template/DateTimeUtils.h#L38-L49

Reply:
updated